includes/admin.form.inc

```diff
@@ -351,22 +354,1275 @@ function islandora_scholar_admin_pdf_form($form, &$form_state) {
+    '#states' => array(
+      'visible' => array(
+        ':input[name="islandora_scholar_xpaths_enabled"]' => array('checked' => TRUE),
+      ),
+    ),
+    '#ajax' => array(
+      'callback' => 'islandora_scholar_xpaths_citeproc_callback',
+      'wrapper' => 'islandora_scholar_xpaths_citeproc_title',
+    ),
+   '#field_suffix' => (!empty($form_state['values']['islandora_scholar_xpaths_demo']) ? _mods_id($form_state['values']['islandora_scholar_xpaths_local_author'], check_plain($form_state['values']['islandora_scholar_xpaths_demo'])) : t("Modify to refresh value (PID for demo object is required).")),
+  ));
+  $form['islandora_scholar_xpaths_citeproc'] += _islandora_scholar_add_variable('islandora_scholar_xpaths_local_container-author', '//mods:mods[1]/mods:relatedItem[@type="host"]/mods:name', array(
+    '#type' => 'textfield',
+    '#title' => t('Container Author'),
+    '#description' => t('Default Value: //mods:mods[1]/mods:relatedItem[@type="host"]/mods:name <br/>XPath to use for the container-author.'),
+    '#states' => array(
+      'visible' => array(
+        ':input[name="islandora_scholar_xpaths_enabled"]' => array('checked' => TRUE),
+      ),
+    ),
+    '#ajax' => array(
+      'callback' => 'islandora_scholar_xpaths_citeproc_callback',
+      'wrapper' => 'islandora_scholar_xpaths_citeproc_title',
+    ),
+   '#field_suffix' => (!empty($form_state['values']['islandora_scholar_xpaths_demo']) ? _mods_id($form_state['values']['islandora_scholar_xpaths_local_container-author'], check_plain($form_state['values']['islandora_scholar_xpaths_demo'])) : t("Modify to refresh value (PID for demo object is required).")),
+  ));
+  $form['islandora_scholar_xpaths_citeproc'] += _islandora_scholar_add_variable('islandora_scholar_xpaths_local_collection-editor', '//mods:mods[1]/mods:relatedItem[@type="series"]/mods:name', array(
+    '#type' => 'textfield',
+    '#title' => t('Collection Editor'),
+    '#description' => t('Default Value: //mods:mods[1]/mods:relatedItem[@type="series"]/mods:name <br/>XPath to use for the collection-editor.'),
+    '#states' => array(
+      'visible' => array(
+        ':input[name="islandora_scholar_xpaths_enabled"]' => array('checked' => TRUE),
+      ),
+    ),
+    '#ajax' => array(
+      'callback' => 'islandora_scholar_xpaths_citeproc_callback',
+      'wrapper' => 'islandora_scholar_xpaths_citeproc_title',
+    ),
+   '#field_suffix' => (!empty($form_state['values']['islandora_scholar_xpaths_demo']) ? _mods_id($form_state['values']['islandora_scholar_xpaths_local_collection-editor'], check_plain($form_state['values']['islandora_scholar_xpaths_demo'])) : t("Modify to refresh value (PID for demo object is required).")),
+  ));
+  $form['islandora_scholar_xpaths_citeproc'] += _islandora_scholar_add_variable('islandora_scholar_xpaths_role_term', 'mods:role/mods:roleTerm', array(
+    '#type' => 'textfield',
+    '#title' => t('Role Term'),
+    '#description' => t('Default Value: mods:role/mods
```

Comment:
(Being afraid of having my comment getting attached to the wrong line, again, let me point out that the following is relative to [L1376](https://github.com/Islandora/islandora_scholar/pull/299/files#diff-0cdd19abe14e51fe9e7f061ec72c86d3R1376).)

Maybe I am seeing this wrong, but would this line not cause _severe_ havoc if a user were to enter "rm / -rf" (or similar) as the path to `pdftotext`? I know that [this is legacy code](https://github.com/Islandora/islandora_scholar/blob/39750e5097f938bc58fe981a48ec9256f593d2ea/includes/admin.form.inc#L226-L238), but I feel we should not leave this as is...

Reply:
OK, will do. This just came in as a complaint to the security group a few days ago so it would make sense to sanitize it.

Reply:
@DonRichards Well, if we sanitise it here, we should probably do this in coordination with the ISRT, lest we implement a different (less powerful) solution than them that they might overlook to modify at a later stage.